packages/main/src/plugin/kubernetes-context-state.spec.ts

```diff
@@ -109,39 +133,48 @@ const apiSender: ApiSenderType = {
   receive: vi.fn(),
 };
 
+const mocks = vi.hoisted(() => {
+  return {
+    makeInformerMock: vi.fn(),
+  };
+});
+
 vi.mock('@kubernetes/client-node', async importOriginal => {
   const actual = await importOriginal<typeof kubeclient>();
   return {
     ...actual,
-    makeInformer: fakeMakeInformer,
+    makeInformer: mocks.makeInformerMock,
   };
 });
 
+// Needs to mock these values to make the backoff much longer than other timeouts, so connection are never retried during the tests
 vi.mock('./kubernetes-context-state-constants.js', () => {
   return {
     connectTimeout: 1,
     backoffInitialValue: 1000,
     backoffLimit: 1000,
-    backoffJitter: 1,
+    backoffJitter: 0,
   };
 });
 
 const originalConsoleDebug = console.debug;
 const consoleDebugMock = vi.fn();
 
 beforeEach(() => {
-  vi.resetAllMocks();
   console.debug = consoleDebugMock;
+  vi.useFakeTimers();
 });
 
 afterEach(() => {
   console.debug = originalConsoleDebug;
+  vi.resetAllMocks();
 });
 
 test('should send info of resources in all reachable contexts and nothing in non reachable', async () => {
+  mocks.makeInformerMock.mockImplementation(fakeMakeInformer);
```

Comment:
vi.mocked(....makeInformer).mockImplementation(fakeMakeInformer);

so you'll have the typechecking